metrics2/metrics2/src/main/java/io/helidon/metrics/MetricsSupport.java

```diff
@@ -182,29 +184,40 @@ private void optionsAll(ServerRequest req, ServerResponse res, Registry registry
 
     static String toPrometheusData(Registry... registries) {
         return Arrays.stream(registries)
+                .filter(r -> !r.empty())
                 .map(MetricsSupport::toPrometheusData)
                 .collect(Collectors.joining());
     }
 
     static String toPrometheusData(Registry registry) {
-        return registry.stream()
-                .sorted(Comparator.comparing(Map.Entry::getKey))
-                .collect(StringBuilder::new,
-                        (sb, entry) -> toPrometheusData(sb, entry.getKey(), entry.getValue()),
-                        StringBuilder::append)
-                .toString();
+        StringBuilder builder = new StringBuilder();
+        Set<String> serialized = new HashSet<>();
+        List<Map.Entry<MetricID, HelidonMetric>> metrics = registry.stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toList());
+        for (Map.Entry<MetricID, HelidonMetric> entry : metrics) {
+            String name = entry.getKey().getName();
+            if (!serialized.contains(name)) {
+                toPrometheusData(builder, entry.getKey(), entry.getValue(), true);
+                serialized.add(name);
+            } else {
+                toPrometheusData(builder, entry.getKey(), entry.getValue(), false);
```

Comment:
The stream is sorted by key (`MetricID`), and therefore primarily by name according to the  `MetricID#compareTo` implementation. So, instead of storing each processed metric name in  a `Set` and checking it each time, we could save the most-recently-processed name in a String (initialized to null) and if the name being processed is different then (and only then) pass `withHelpType` as `true`. And assigning the current name to the variable holding the previously-processed one (even it's the same value) is pretty cheap, so the code in the `forEach` could be something like:
```
// inside the forEach
String name = entry.getKey().getName();
toPrometheusData(builder, entry.getKey(), entry.getValue(), !name.equals(previousName);
previousName = name;
```
which is a bit simpler and quite a bit faster. Granted, this does depend on the ordering provided by `MetricID#compareTo` always giving preference to the name before the tag (or other stuff that might be added to `MetricID` in the future) so that, when sorted, `MetricID`s are grouped by name. But that seems fairly reliable.